src/Sentry/Sentry.csproj

```diff
@@ -23,21 +23,24 @@
   <Import Project="Platforms\iOS\Sentry.iOS.props" Condition="'$(TargetPlatformIdentifier)' == 'ios' Or '$(TargetPlatformIdentifier)' == 'maccatalyst'" />
   <Import Project="Platforms\Native\Sentry.Native.props" Condition="('$(TargetFramework)' == 'net7.0' Or '$(TargetFramework)' == 'net8.0') And ($([MSBuild]::IsOSPlatform('Linux')) Or $([MSBuild]::IsOSPlatform('Windows')))" />
 
-  <!--
-    Ben.Demystifier is compiled directly into Sentry.
-    Note: It uses Microsoft.Bcl.AsyncInterfaces, which we get transitively from System.Text.Json.
-  -->
-  <ItemGroup>
-    <Compile Include="..\..\modules\Ben.Demystifier\src\**\*.cs">
-      <Link>%(RecursiveDir)\%(Filename)%(Extension)</Link>
-    </Compile>
-    <Compile Remove="..\..\modules\Ben.Demystifier\**\obj\**" />
-  </ItemGroup>
+  <Target Name="IncludeBenDemystifier" BeforeTargets="BeforeBuild" Condition="'$(PublishAot)' != 'true'">
+    <!--
+      Ben.Demystifier is compiled directly into Sentry.
+      Note: It uses Microsoft.Bcl.AsyncInterfaces, which we get transitively from System.Text.Json.
+    -->
+    <ItemGroup>
+      <Compile Include="..\..\modules\Ben.Demystifier\src\**\*.cs">
+<!-- TODO: Work out why we get ' Error NETSDK1022 : Duplicate 'Compile' items were included.' if we uncomment this -->
+<!--        <Link>%(RecursiveDir)\%(Filename)%(Extension)</Link>-->
```

Comment:
Can this be addressed before merging so that it doesn't get lost? Or has this always been the case and it has nothing to do with this PR?

Reply:
No, I introduced that. It took me about a day to find that and I don't understand it at all. Logically, that section of the csproj file shouldn't be included when `PublishAot == true`.. but it is. I assume it relates to some arcane details of MS Build that I don't know about and I couldn't find any explanation in my googling attempts. 
1. I'd like to know what the implication of commenting this out would be. 
2. I'd like to know why we need to comment it out to get things to build properly. 
   - Possibly with a minimum reproducible example, we could get an answer from the team that works on MSBuild...

Reply:
Found a workaround to this.